NetworkManager-openvpn :  update to 1.8.18 + gtk4 support

hsnfirdaus opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES**
<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->
#### Local build testing
- I built this PR locally for my native architecture, (x86_64-LIBC)

kruceter reviewed (changes requested):
Replace `sed` in the template with `vsed`.

Change your commit message header to:

`NetworkManager-openvpn: update to 1.8.18, add gtk4 support.`

Rebase the master branch against your own instead of merging.

kruceter commented:
@hsnfirdaus, squash these two commits together. Do not forget to alter the commit message the way I asked you to.

hsnfirdaus commented:
@kruceter should i resolve this conflict too?

kruceter commented:
> should i resolve this conflict too?

Rebase your branch with recent changes from master.

```
git checkout master
git pull --rebase upstream master
git rebase master nm-openvpn
```

Push again after that.

hsnfirdaus commented:
> ```
> git rebase master nm-openvpn
> ```

@kruceter how about this message when i run it: 
```
...
Rebasing (131/160618)
Rebasing (132/160618)
Auto-merging .gitignore
CONFLICT (add/add): Merge conflict in .gitignore
error: could not apply 16ae7bdf63... Added a .gitignore file to ignore default created dirs.
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 16ae7bdf63... Added a .gitignore file to ignore default created dirs.
```

kruceter commented:
Try to skip them all, @hsnfirdaus.

paper42 commented:
That's an issue in your checkout because you modified .gitignore for some reason, but it's not in a commit. Abort this, make sure you don't need your local changes (git diff), remove them (git restore .) and try again.

hsnfirdaus commented:
> That's an issue in your checkout because you modified .gitignore for some reason, but it's not in a commit. Abort this, make sure you don't need your local changes (git diff), remove them (git restore .) and try again.

Well i ended up this conflict by re clone my repo and rebase...

Thank you @paper42 @kruceter

paper42 commented on srcpkgs/NetworkManager-openvpn/template:
```diff
@@ -1,27 +1,27 @@
 pre_configure() {
 	autoreconf -fi
-}
+}
```
don't delete the end of the line here

paper42 commented on srcpkgs/NetworkManager-openvpn/template:
```diff
@@ -1,27 +1,27 @@
 maintainer="Enno Boland <gottox@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://wiki.gnome.org/Projects/NetworkManager"
 distfiles="${GNOME_SITE}/${pkgname}/${version%.*}/${pkgname}-${version}.tar.xz"
```
```suggestion
distfiles="${GNOME_SITE}/NetworkManager-openvpn/${version%.*}/NetworkManager-openvpn-${version}.tar.xz"
```
